### ISSUE TITLE
Investigate new `sort_child_properties_last` diagnostics when relaxed

### ISSUE THREAD

**Issue by pq** [dart-lang/linter#2713]
#2648 introduces new diagnostics to libraries in (at least) 20 internal packages (the SDK roller stopped trying to add exceptions at 20).  We'll need to investigate if these are false positives or if we need to coordinate fixes before the next roll.

**Comment by pq** [comment]
Example breakage:

```dart
  @override
  Widget build(BuildContext context) => Scaffold(
      appBar: AppBar(
        brightness: widget.brightness,
        leading: IconButton(
          icon: const BackButtonIcon(), // <= LINT
          tooltip: MaterialLocalizations.of(context).backButtonTooltip,
          onPressed: () {
             ...
          },
          key: backButton,
        ),
  ...
```

**Comment by pq** [comment]
Good to know.  Given the impact, we'll have to come up with a plan for migration (or reconsider the additional diagnostics).

Thanks!

/fyi @jacob314

**Comment by jacob314** [comment]
Happy to help migrate. Perhaps it would be worth it to add a quick fix to help migrate? Moving all violating properties to after the last non-child & non closure property would probably be a fine default.
In the example given, `backgroundColor` should be at the top of the method next to the `brightness` property. As is it is, it takes a while to spot what widget the backgroundColor is associated with.

**Comment by pq** [comment]
> Perhaps it would be worth it to add a quick fix to help migrate?

FWIW there is a fix but I expect it will need to be updated w/ the new and improved semantics.
